product.py

```diff
@@ -52,20 +52,30 @@ def _quantity_locations(cls, name=None):
                 warehouses = [Location(warehouse_id)]
             elif config.warehouse_quantity == 'all':
                 warehouses = Location.search([('type', '=', 'warehouse')])
+
             if warehouses:
-                location_ids = [w.storage_location.id for w in warehouses
-                    if w.storage_location]
+                location_ids = [w.id for w in warehouses]
         return location_ids
 
     @classmethod
     def get_quantity(cls, products, name):
-        with Transaction().set_context(locations=cls._quantity_locations(name)):
-            return super().get_quantity(products, name)
+        context = Transaction().context
+
+        if not context.get('locations'):
+            with Transaction().set_context(locations=cls._quantity_locations(name),
+                    with_childs=True):
```

Comment:
In case has multiples warehouses (len(location_ids) > 1), [with_childs is False](https://github.com/tryton/tryton/blob/11bac062f7d32d894df38a74e4b9d31b4fa7c260/modules/stock/move.py#L89)

In this case, force with_childs to True